### PR TITLE
add python-eccodes recipe 2nd try

### PR DESCRIPTION
**Pull request by kmuehlbauer**
Resending removed recipe (see #747)

### CODE REVIEW

File: recipes/python-eccodes/build.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+cmake $src_dir \
+         -DCMAKE_INSTALL_PREFIX=$PREFIX \
+         -DENABLE_JPG=1 \
+         -DENABLE_NETCDF=1 \
+         -DENABLE_PNG=1 \
+         -DENABLE_PYTHON=1 \
+         -DENABLE_FORTRAN=0
+
+make
+eval ${LIBRARY_SEARCH_VAR}=$PREFIX/lib
```
Comment (jakirkham) [inline recipes/python-eccodes/build.sh]: Is there something else that is suppose to be on this line?

Comment (kmuehlbauer) [inline recipes/python-eccodes/build.sh]: The test was there, I'll remove it.

Comment (jakirkham) [inline recipes/python-eccodes/build.sh]: Not trying to rehash the whole thing, but why did it get removed?

**Comment by kmuehlbauer** [comment]
@ocefpaf appveyor runs as seen before, although skipped. I also split the skip. What else did you test?

**Comment by ocefpaf** [comment]
> @ocefpaf appveyor runs as seen before, although skipped. I also split the skip. What else did you test?

The problem here was not `python-eccodes`, but something else. I suspect conda-build. Can you try the change in `appveyor.yml` below?

https://github.com/conda-forge/staged-recipes/pull/745/files

File: appveyor.yml [hunk 2]
```diff
@@ -76,6 +76,7 @@ install:
     - cmd: conda install --yes --quiet obvious-ci conda-build-all
     - cmd: conda install --yes --quiet conda-forge-build-setup
     - cmd: run_conda_forge_build_setup
+    - cmd: conda install --yes --quiet conda-build=1.20.0
```
Comment (ocefpaf) [inline appveyor.yml]: If this is proven we should not merge this, but fix it in [`conda-forge-build-setup`](https://github.com/conda-forge/conda-forge-build-setup-feedstock/blob/master/recipe/run_conda_forge_build_setup_win#L12) instead.

Comment (ocefpaf) [inline appveyor.yml]: We should also ping @msarahan who might help us figure out if the latest `conda-build` is broken or if we are doing something wrong.

**Comment by kmuehlbauer** [comment]
Goodness gracious! We have windows builds! I have no idea what's going on. Have to leave this until the sun comes from the east again...

**Comment by ocefpaf** [comment]
> Goodness gracious! We have windows builds! 

That is quite the wishful thinking :smile_cat: 

The only thing that is happening in `AppVeyor` is that the skip is now being respected. (You don't see skipped Windows builds on the feedstocks b/c `conda-smithy` does not create the `AppVeyor` configuration when `skip: True  # [win]` is present.)

@msarahan can you help us debug what is going on here? @kmuehlbauer had to downgrade `conda-build` to get the `skip: True  # [win]` to work properly.

**Comment by kmuehlbauer** [comment]
So, it's this :sunny:, needing :sunglasses: 

What else can we do with this. Should I test/check something more, to trace down the issue?

**Comment by kmuehlbauer** [comment]
> That is quite the wishful thinking :smile_cat:

Yes, but isn't conda-forge doing magic things all the time... :joy_cat: 

> The only thing that is happening in AppVeyor is that the skip is now being respected. (You don't see skipped Windows builds on the feedstocks b/c conda-smithy does not create the AppVeyor configuration when skip: True # [win] is present. But the AppVeyor configuration is always present in staged-recipes.)

Thanks for clarifying this. I was wondering why this happens.

**Comment by ocefpaf** [comment]
> What else can we do with this. Should I test/check something more, to trace down the issue?

@msarahan any ideas why downgrading `conda-build` solved the issue? With the latest `conda-build` the `skip: True # [win]` ignored conda tries to build it on Windows.

@kmuehlbauer meanwhile let's put a PR to [`conda-forge-build-setup`](https://github.com/conda-forge/conda-forge-build-setup-feedstock/blob/master/recipe/run_conda_forge_build_setup_win#L12) pinning the whole matrix to `conda-build=1.20.0`. (If you cannot do it I will do it later today.)

ping @pelson

**Comment by kmuehlbauer** [comment]
@ocefpaf Got confused with the PR, should be OK now.

**Comment by ocefpaf** [comment]
> @ocefpaf Got confused with the PR, should be OK now.

Thanks a lot for doing that @kmuehlbauer! As soon as we hear from others (and if https://github.com/conda-forge/conda-forge-build-setup-feedstock/issues/7 is merged) let's revert the changes here and merge this one.

**Comment by ocefpaf** [comment]
@kmuehlbauer finally! :tada:

**Comment by jakirkham** [comment]
Looks like this [problem]( https://travis-ci.org/conda-forge/staged-recipes/jobs/137225997#L281-L283 ) has re-emerged. My suspicion is we need the same `conda-build` pinning. Though this is only a guess.

```
conda.resolve.NoPackagesFound:  Packages missing in current win-32 channels: 
  - jasper
  - eccodes 0.13.0
```

**Comment by ocefpaf** [comment]
@jakirkham please do not remove the package from master. I need it there to complete the testing and pinnings.

**Comment by jakirkham** [comment]
Leaving it alone.

**Comment by ocefpaf** [comment]
> Leaving it alone.

Thanks! Note that your input and help is very welcomed! However, I cannot debug this if the packages are removed. I also do not I want to force people to re-submit their PRs.

We need to fix this and note sweep it under the rug. The process is slow, tedious, and take a lot of trial-and-errors, so some patience is required :wink:

**Comment by jakirkham** [comment]
This makes @kmuehlbauer my second guinea pig for the new form of team additions ( https://github.com/conda-forge/staged-recipes/pull/733 ). Looks like everything went ok AFAICT. Please let me know if something didn't.

**Comment by kmuehlbauer** [comment]
@jakirkham 

I got two emails, one from conda-forge:

Subject: ```Automatic conda-forge administrator added you to the conda-forge team python-eccodes```
Body: ```You’ve been added to the python-eccodes team for the conda-forge organization.  python-eccodes has 1 member and gives *push* access to 1 conda-forge repository.

View python-eccodes: https://github.com/orgs/conda-forge/teams/python-eccodes

Read more about team permissions here: https://help.github.com/articles/what-are-the-different-access-permissions ```

The second is from github about notification subscription. So, seems your additions ran fine.

**Comment by jakirkham** [comment]
Excellent. Good to hear it worked. Hoping this cuts down the effects of rate limiting a bit more here. We have some more ideas, but this is a good start already.

**Pull request merged**